distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso

```diff
@@ -110,48 +120,632 @@ type Array
+             [1, 2, 3].to_array.at 1 == 2
 
        > Example
          Get the last element of an array.
 
-             [1,2,3].to_array.at -1
+             [1, 2, 3].to_array.at -1 == 3
     at : Integer -> Any ! Index_Out_Of_Bounds
     at self index = @Builtin_Method "Array.at"
 
-    ## Gets the length of the array this.
+    ## Returns the number of elements stored in this array.
 
        > Example
-         Getting the length of an array.
+         Checking the length of an array.
 
-             [1,2,3].to_array.length
+             [1, 2, 3, 4].to_array.length
     length : Integer
     length self = @Builtin_Method "Array.length"
 
-    ## Sorts the Array.
-
-      Arguments:
-      - comparator: A comparison function that takes two elements and returns
-        an Ordering that describes how the first element is ordered with
-        respect to the second.
-
-      > Example
-        Getting a sorted array of numbers.
-
-            [3,2,1].to_array.sort
-    sort : (Any -> Any -> Ordering) -> Array
-    sort self comparator=(Ordering.compare _ _) =
-        self.sort_builtin comparator
-
     ## Identity.
        This method is implemented purely for completeness with the runtime's
        primitive array protocol.
     to_array : Array
-    to_array self = @Builtin_Method "Array.to_array"
+    to_array self = self
+
+    #
+    # For compatibility with Vector
+    #
+
+    ## Sort the array.
+
+       Arguments:
+       - order: The order in which the array elements are sorted.
+       - on: A projection from the element type to the value of that element
+         being sorted on.
+       - by: A function that compares the result of applying `on` to two
+         elements, returning an Ordering to compare them.
+
+       By default, elements are sorted in ascending order, using the comparator
+       acquired from each element. A custom compare function may be passed to
+       the sort method.
+
+       This is a stable sort, meaning that items that compare the same will not
+       have their order changed by the sorting process.
+
+       ! Computational Complexity
+         The complexities for this sort are:
+         - *Worst-Case Time:* `O(n * log n)`
+         - *Best-Case Time:* `O(n)`
+         - *Average Time:* `O(n * log n)`
+         - *Worst-Case Space:* `O(n)` additional
+
+       ? Implementation Note
+         The sort implementation is based upon an adaptive, iterative mergesort
+         that requires far fewer than `n * log(n)` comparisons when the array
+         is partially sorted. When the array is randomly ordered, the
+         performance is equivalent to a standard mergesort.
+
+         It takes equal advantage of ascending and descending runs in the array,
+         making it much simpler to merge two or more sorted arrays: simply
+         concatenate them and sort.
+
+       > Example
+         Sorting an array of numbers.
+
+             [5, 2, 3, 45, 15].to_array.sort == [2, 3, 5, 15, 45].to_array
```

Comment:
It would be cool if we had a better way of writing arrays then `[1, 2, 3].to_array` in all of those examples.

Reply:
Alas, `.to_array` is what we use even in tests. Luckily there shall be less and less reasons to recognize/convert to `Array`.